src/ProjectOrigin.PedersenCommitment/Ristretto/Oracle.cs

```diff
@@ -0,0 +1,48 @@
+namespace ProjectOrigin.PedersenCommitment.Ristretto;
+using System.Text;
+
+class Oracle
+{
+
+    private List<byte[]> messages = new List<byte[]>();
+
+    public Oracle(byte[] label)
+    {
+        messages.Add(label);
+    }
+
+    public void Domain(String domain)
+    {
+        var bytes = Encoding.UTF8.GetBytes("test");
+        messages.Add(bytes);
+    }
+
+
+    public void Add(params Point[] points)
+    {
+
+        foreach (Point p in points)
+        {
+            messages.Add(p.Compress()._bytes);
+        }
+    }
+
+    public Scalar Challenge()
```

Comment:
Here one could also use a label as domain separation (e.g. between to subsequent challenges).
```
public Scalar Challenge(String label)
{
var byte_label = Encoding.UTF8.GetBytes(label);
messages.add(byte_label)
//Then do the challenge computation
}